src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        Date dateFromDate = new Date();
+        Date dateToDate = new Date();
+        Date date = new Date();
+        StringBuilder builder = new StringBuilder("Report for period ")
+                .append(dateFrom).append(" - ").append(dateTo);
+        try {
+            dateFromDate = new SimpleDateFormat("dd.MM.yyyy").parse(dateFrom);
+        } catch (ParseException e) {
+            System.out.println("Date from parsed error: " + dateFrom);
+        }
+        try {
+            dateToDate = new SimpleDateFormat("dd.MM.yyyy").parse(dateTo);
+        } catch (ParseException e) {
+            System.out.println("Date to parsed error: " + dateTo);
+        }
+        for (String name : names) {
+            builder.append(System.lineSeparator()).append(name).append(" - ");
+            int totalSallery = 0;
+            for (String dataRow : data) {
+                if (dataRow.isEmpty()) {
+                    continue;
+                }
+                String[] dataRowArray = dataRow.split(" ");
+                try {
+                    date = new SimpleDateFormat("dd.MM.yyyy").parse(dataRowArray[0]);
```

Comment:
[Any magic numbers should be constants.](https://mate-academy.github.io/jv-program-common-mistakes/java-core/exceptions/salary-info#any-magic-numbers-should-be-constants) Also, I recommend you make a constant separator and header.